docs/data/data-grid/row-selection/row-selection.md

```diff
@@ -36,6 +36,12 @@ In the demo below only rows with quantity above 50,000 can be selected:
 
 {{"demo": "DisableRowSelection.js", "bg": "inline"}}
 
+## Row selection with filtering
+
+By default, when the rows are filtered the selection is cleared from the rows that do not meet the filter criteria. To keep those rows selected even when they are not visible, set the `keepNonExistentRowsSelected` prop.
```

Comment:
Formatting nitpick: one sentence per line please—makes it easier to do line-by-line reviews in bigger docs/PRs. Also, don't be afraid of contractions! 😁

```suggestion
By default, when the rows are filtered the selection is cleared from the rows that don't meet the filter criteria. 
To keep those rows selected even when they're not visible, set the `keepNonExistentRowsSelected` prop.
```